src/renderers/WebGLRenderer.js

```diff
@@ -1882,91 +1881,15 @@ function WebGLRenderer( parameters ) {
 
 			}
 
-			// refresh uniforms common to several materials
+			// refresh uniforms common to materials
 
 			if ( fog && material.fog ) {
 
 				refreshUniformsFog( m_uniforms, fog );
```

Comment:
What about `refreshUniformsFog()`? Could this also be replaced?

Reply:
I think that no for now. `fog` and `lights` come out of the pattern of material properties.

Reply:
> fog and lights come out of the pattern of material properties.

Um, I'm not sure I understand this. Can you please explain in more detail? 😇

Reply:
I was based on the signature of the function, I forgot that I add `scene` parameter in the function. I add this feature 👍